### ISSUE TITLE
Pattern notation leads to `Error: The variable pat was not found in the current environment.`

### ISSUE THREAD

**Issue by Janno** [coq/coq#8672]
#### Version
8.8.1, master

#### Description of the problem

The following code produces a somewhat cryptic (unless one knows about the encoding of patterns) error message. Additionally, I don't really know what behavior to expect from this notation. Should it work at all? I think so, but my intuition here is usually wrong.

```coq
Notation "'plet' x := e 'in' t" :=
  ((fun H => let x := id H in t) e) (at level 0, x pattern).
Definition bla := plet (pair x y) := pair 1 2 in x.
```

**Comment by Janno** [comment]
Small update: the problem goes away when I define write `let 'x := ..` instead of `let x := ..`. I suppose the original notation should not have been accepted after all.

**Comment by herbelin** [comment]
The behavior of your first example was a bug. I fixed it in #8676. Thanks a lot for reporting!
